libvast/src/system/query_supervisor.cpp

```diff
@@ -48,26 +48,27 @@ query_supervisor_actor::behavior_type query_supervisor(
   // Ask master for initial work.
   self->send(master, atom::worker_v, self);
   return {
-    [=](const expression&, const query_map& qm,
+    [=](const expression& expr,
+        const std::vector<std::pair<uuid, partition_actor>>& qm,
         const index_client_actor& client) {
       VAST_DEBUG(self, "got a new query for", qm.size(),
                  "partitions:", get_ids(qm));
       VAST_ASSERT(!qm.empty());
       VAST_ASSERT(self->state.open_requests.empty());
-      for (auto& kvp : qm) {
-        auto& id = kvp.first;
-        auto& evaluators = kvp.second;
-        VAST_DEBUG(self, "asks", evaluators.size(),
-                   "EVALUATOR actor(s) for partition", id);
-        self->state.open_requests.emplace(id, evaluators.size());
-        for (auto& evaluator : evaluators)
-          self->request(evaluator, caf::infinite, client).then([=](atom::done) {
+      for (auto& [id, partition] : qm) {
+        self->state.open_requests.emplace(id, 1 /*qm.size()*/);
```

Comment:
Even in the old design I don't think there could be more than 1. I think `open_requests` can be replaced by a count, and we can refactor to use `fan_out_request` when we drop compatibility with CAF 0.17.

Reply:
Good point, updated.